collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/SavePointTest.kt

```diff
@@ -0,0 +1,239 @@
+        assertThat(auditLog[3].get("event"), equalTo("question"))
+        assertThat(auditLog[4].get("event"), equalTo("form save"))
+        assertThat(auditLog[5].get("event"), equalTo("form exit"))
+    }
+
+    @Test
+    fun whenEditing_savePointIsCreatedWhenLeavingTheApp() {
+        // Create instance
+        rule.setUpProjectAndCopyForm("two-question-audit.xml")
+            .fillNewForm("two-question-audit.xml", "Two Question")
+            .fillOutAndSave(
+                AppClosedPage(),
+                FormEntryPage.QuestionAndAnswer("What is your name?", "Pasquale"),
+                FormEntryPage.QuestionAndAnswer("What is your age?", "52")
+            )
+
+        // Create save point
+        rule.editForm("two-question-audit.xml", "Two Question")
+            .clickGoToStart()
+            .answerQuestion("What is your name?", "Alexei")
+            .let { simulateProcessRestore() }
+
+        // Edit instance and check save point is loaded
+        rule.editForm("two-question-audit.xml", "Two Question")
+            .assertText("Alexei")
+            .pressBack(FormEntryPage("Two Question"))
+            .assertQuestion("What is your name?")
+            .pressBack(SaveOrIgnoreDialog("Two Question", AppClosedPage()))
+            .clickSaveChanges()
+
+        // Check audit log
+        val auditLog = StorageUtils.getAuditLogForFirstInstance()
+        assertThat(auditLog.size, equalTo(12))
+
+        assertThat(auditLog[5].get("event"), equalTo("form resume"))
+        assertThat(auditLog[6].get("event"), equalTo("jump"))
+        // Question event not logged - possibly a problem
+
+        assertThat(auditLog[7].get("event"), equalTo("form resume"))
+        assertThat(auditLog[8].get("event"), equalTo("jump"))
+        assertThat(auditLog[9].get("event"), equalTo("question"))
+        assertThat(auditLog[10].get("event"), equalTo("form save"))
+        assertThat(auditLog[11].get("event"), equalTo("form exit"))
+    }
+
+    @Test
+    fun blankFormSavePointIsNotUsedWhenEditingInstance() {
+        // Create instance
+        rule.setUpProjectAndCopyForm("two-question-audit.xml")
+            .fillNewForm("two-question-audit.xml", "Two Question")
+            .fillOutAndSave(
+                AppClosedPage(),
+                FormEntryPage.QuestionAndAnswer("What is your name?", "Pasquale"),
+                FormEntryPage.QuestionAndAnswer("What is your age?", "52")
+            )
+
+        // Create save point for blank form
+        rule.fillNewForm("two-question-audit.xml", "Two Question")
+            .answerQuestion("What is your name?", "Alexei")
+            .let { simulateProcessRestore() }
+
+        // Check editing instance doesn't load save point
+        rule.editForm("two-question-audit.xml", "Two Question")
+            .assertText("Pasquale")
+            .assertTextDoesNotExist("Alexei")
+    }
+
+    @Test
+    fun instanceSavePointIsNotUsedWhenFillingBlankForm() {
```

Comment:
It's because the app was killed when editing a form right? If it happened on filling a new form the result would be different. If so it would be good to improve the title or add a comment to make it clear.